### ISSUE TITLE
layer_shell does not obtain keyboard focus in certain corner cases

### ISSUE THREAD

**Issue by kennylevinsen** [swaywm/sway#4407]
Experienced using https://github.com/kennylevinsen/wldash at commit 9ac29b2a94e406bfa6442bb549b773b4a073ef19, which creates a top layer shell surface that is keyboard interactive.

# To reproduce

Note that this issue is finicky, and depends on what applications are currently running and have focus.

1. Start sway with the following configuration:
```
bar {
    position bottom
    mode hide
    modifier $mod
    ...
}

$mod+t exec wldash
```
2. Start a terminal emulator, such as alacritty
3. Launch wldash with $mod+t
4. Press escape

# Expected result

wldash receives escape and terminates

# Actual result

wldash does not have keyboard focus until focus is changed, causing sway to acknowledge wldash's keyboard_interactive top layer_shell surface.

This seems to be due to interaction with swaybar, as removing the bar resolves the issue.

**Comment by ddevault** [comment]
Can you test this again now that #4467 has been merged?

**Comment by kennylevinsen** [comment]
Verified fixed. Thanks!
